### ISSUE TITLE
No hay forma de prioritizar flags activas

### ISSUE THREAD

**Issue by danielrossyamitrano** [ludica-squamata/mano-gift#154]
[Al habilitar nuevamente el diálogo de pruebas](https://github.com/ludica-squamata/mano-gift/commit/c8dccbd00759071e9b1cd764be3e9712b6d18e71), me di cuenta de que no hay forma de prioritizar qué flags disparan diálogos, o en qué orden, si están todas activas al mismo tiempo. Si bien esto se puede solucionar alternando las flags de manera secuencial, debería haber un sistema general de prioritizar no solo flags de diálogos, sino flags en general, en caso de que haya múltiples flags activas (lo cual es altamente probable).

Este issue, originado por un conflicto en los diálogos, se relaciona con el #108, pero abro un issue nuevo para no modificar ese y extenderlo todavía más.

**Comment by danielrossyamitrano** [comment]
Una forma de ordenarlas podría ser por orden de llegada. En el ejemplo de los diálogos, el primero es el de prueba, pues su flag se establece en el script. Las flags para los items se añaden cuando el item en cuestión está en poder del héroe, a lo cual habría un orden si esto se respetara. Sin embargo, como GameState es basicamente un dict, no es posible tal orden por default. Tal vez este issue llame a un reworking de GameState, que ya de por sí avisa en todos lados que "la implementación puede cambiar en el futuro".

**Comment by einacio** [comment]
Las flags en si mismas no tienen prioridad,  por definicion solo estan o no estan.
El orden y la prioridad van definidas en el codigo que las utiliza, porque ahi si podes tener logica.

La misma flag puede tener diferente prioridad en dos eventos/dialogos/objetos distintos.

Por ejemplo, matar al dragon que quemas los campos es muy importante para el lord que cobra impuestos, e importante tambien para el granjero.
Pero salvar a la hija del granjero es mas importante para el granjero, y para el lord es solo una palmadita, "gracias por cuidar a mis subditos".

El orden en que se checkean importa, no es prioridad implicita

29 mar. 2021 5:00:07 Daniel ***@***.***>:

**Comment by danielrossyamitrano** [comment]
> El orden en que se checkean importa, no es prioridad implicita

A eso me refiero con "prioridad".

**Comment by danielrossyamitrano** [comment]
Mejor dicho, resulelve (al menos las flags de diálogo) por orden alfabético según cómo se llama el json del diálogo en cuestión. Lo que no me parece correcto.

**Comment by danielrossyamitrano** [comment]
Lo cierro por inválido. El issue no esta en las flags en sí ni en GameState, sino en la función DialogCircularMenu.is_possible().

**Comment by danielrossyamitrano** [comment]
No, lo reabro hasta que los diálogos tengas algun tipo de órden por relevancia.

**Comment by danielrossyamitrano** [comment]
Una cosa que se me ocurre ahora es que los diálogos no tengan una "relevancia" o "prioridad" fija que los ordene, sino una posibilidad cambiante de que un npc sepa sobre o te hable sobre algo. Por ejemplo, en un mapa donde no hay bananas, no habría monos, por lo que la posibilidad de que un npc te responda sobre el tema es 0%. En cambio, si hay camas en el mapa, el npc podria decidir que alguna de esas camas es la de él, y si le pedis que vaya a "su" cama, "sabría" de qué estás hablando. 

Luego, si dos díalogos tienen la misma posibilidad, ahi sí se podrián elegir aleatoriamente, pero creo que tal coincidencia sería más bien rara.
